src/sentry/integrations/msteams/card_builder/__init__.py

```diff
@@ -22,348 +18,20 @@
-        return f"#{assignee.slug}"
-
-    return None
-
-
-def build_group_title(group):
-    # TODO: implement with event as well
-    ev_metadata = group.get_event_metadata()
-    ev_type = group.get_event_type()
-
-    if ev_type == "error" and "type" in ev_metadata:
-        text = ev_metadata["type"]
-    else:
-        text = group.title
-
-    link = group.get_absolute_url(params={"referrer": "msteams"})
-
-    title_text = f"[{text}]({link})"
-    return {
-        "type": "TextBlock",
-        "size": "Large",
-        "weight": "Bolder",
-        "text": title_text,
-        "wrap": True,
-    }
-
-
-def build_group_descr(group):
-    # TODO: implement with event as well
-    ev_type = group.get_event_type()
-    if ev_type == "error":
-        ev_metadata = group.get_event_metadata()
-        text = ev_metadata.get("value") or ev_metadata.get("function")
-        return {
-            "type": "TextBlock",
-            "size": "Medium",
-            "weight": "Bolder",
-            "text": text,
-            "wrap": True,
-        }
-    else:
-        return None
-
-
-def build_rule_url(rule, group, project):
-    org_slug = group.organization.slug
-    project_slug = project.slug
-    rule_url = f"/organizations/{org_slug}/alerts/rules/{project_slug}/{rule.id}/details/"
-    return absolute_uri(rule_url)
-
-
-def build_group_footer(group, rules, project, event):
-    # TODO: implement with event as well
-    image_column = {
-        "type": "Column",
-        "items": [
-            {
-                "type": "Image",
-                "url": absolute_uri(get_asset_url("sentry", "images/sentry-glyph-black.png")),
-                "height": "20px",
-            }
-        ],
-        "width": "auto",
-    }
-
-    text = f"{group.qualified_short_id}"
-    if rules:
-        rule_url = build_rule_url(rules[0], group, project)
-        text += f" via [{rules[0].label}]({rule_url})"
-        if len(rules) > 1:
-            text += f" (+{len(rules) - 1} other)"
-
-    text_column = {
-        "type": "Column",
-        "items": [{"type": "TextBlock", "size": "Small", "weight": "Lighter", "text": text}],
-        "isSubtle": True,
-        "width": "auto",
-        "spacing": "none",
-    }
-
-    date_ts = group.last_seen
-    if event:
-        event_ts = event.datetime
-        date_ts = max(date_ts, event_ts)
-
-    date = date_ts.replace(microsecond=0).isoformat()
-    date_text = f"{{{{DATE({date}, SHORT)}}}} at {{{{TIME({date})}}}}"
-    date_column = {
-        "type": "Column",
-        "items": [
-            {
-                "type": "TextBlock",
-                "size": "Small",
-                "weight": "Lighter",
-                "horizontalAlignment": "Center",
-                "text": date_text,
-            }
-        ],
-        "width": "auto",
-    }
-
-    return {"type": "ColumnSet", "columns": [image_column, text_column, date_column]}
-
-
-def build_group_actions(group, event, rules, integration):
```

Comment:
This is what `MSTeamsIssueMessageBuilder.get_issue_actions` replaces. Very lengthy function to build 3 buttons.
All of them have a common pattern - if toggled create a submit button that will undo the toggled action, otherwise they reveal a card with choices on how to perform the action. Extracted out this common pattern into `MSTeamsIssueMessageBuilder. create_issue_action_block`